src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -95,6 +104,50 @@ public static Task<List<string>> ReadRequestAndSendResponseAsync(Socket server,
             return lines;
         }
 
+        public static async Task<bool> WebSocketHandshakeAsync(Socket s, StreamReader reader, StreamWriter writer)
+        {
+            string serverResponse = null;
+            string currentRequestLine;
+            while (!string.IsNullOrEmpty(currentRequestLine = await reader.ReadLineAsync().ConfigureAwait(false)))
+            {
+                string[] tokens = currentRequestLine.Split(new char[] { ':' }, 2);
+                if (tokens.Length == 2)
+                {
+                    string headerName = tokens[0];
+                    string headerValue = tokens[1].Trim();
```

Comment:
nit: the headerValue string declaration and assignment could be moved within the clause below